### ISSUE TITLE
Please restore combined datasets (perhaps as releases)

### ISSUE THREAD

**Issue by jgoerzen** [cipriancraciun/covid19-datasets#42]
Hi,

I understand the issue with the combined dataset.  However, this was very helpful.

https://docs.github.com/en/github/managing-large-files/distributing-large-binaries discusses options for this.  Over at https://github.com/jgoerzen/covid19db I generate a dataset that aggregates from yours and some others.  I use a Github Action that automatically publishes it to Github as a release; see https://github.com/jgoerzen/covid19db/blob/master/.github/workflows/build.yml

It should be fairly easy to do that here.

Also, locations-diff.tsv is important for some analysis, and it would be nice to see it back as well.

Thanks,

John

**Comment by cipriancraciun** [comment]
Thanks for the report!

I'll investigate today a way to bring it back, although perhaps outside the GitHub environment, by example uploading it on a "plain" HTTP server.

----

The issue is not only with the size of the individual file, which even compressed approaches the 100MiB limit, but also with the overall size of the entire repository.  At the moment GitHub reports it has over 75 GiB, therefore I estimate in at most 2 months it will reach the 100 GiB limit, at which point I'll be in trouble...

At the moment there are two solutions for large files:
* as you've said releases, which it seems imposes no limit;
* Git LFS (Large File Storage), https://docs.github.com/en/github/managing-large-files/versioning-large-files, which unfortunately for a free account is limited to 1 GiB both in storage and bandwidth;  (thus out of the question for my use-case;)

Getting back to the releases feature it has a few drawbacks:
* first, given that I re-generate the datasets daily, it will quickly amount hundreds of "releases", thus becoming a mess;
* secondly, I'll have to implement it as a separate step in the processing pipeline;  (I can't use GitHub actions, because the time required to "process" the current data will quickly consume all the free allocations;  for example at the moment the processing takes about 2 hours on a 8 core system with 32 GiB of RAM;)

----

Thus based on the above observations, I must rethink my distribution model for output files, which it seems doesn't match anymore the "self-contained" GitHub repository.  :(

**Comment by jgoerzen** [comment]
Thanks for thinking about it!  By the way, you can have a single "release" which you keep overwriting every day.  That's the approach I use for covid19db.  This has the effect of not keeping historical datasets around, which is a drawback but probably the most practical option at the moment.

I did adapt covid19db to download the various constituent datasets and combine them, and extract locations-diff.tsv from a previous commit, so I'm unblocked for now but it's a bit sub-optimal.

As far as I can tell, GitHub actions have no particular limits for public repos on the free tier.  But I may be missing something?

FWIW, the Rust code I'm using in covid19db runs in about 10 minutes in GitHub Actions.  The majority of that time actually goes into compiling dependencies.  The most time-consuming part of the actual data ingestion is in processing your datasets.  Running the Rust code in debug mode (without optimizations) causes it to compile faster but run slower.  It's probably not worth the effort, but I would surmise that redoing your processing pipeline in a compiled language like Rust or Go, targeting Sqlite directly with commits in long-running transactions, and then exporting the other formats from there would dramatically reduce the preparation time.

Anyhow, I have made sure to credit your project in covid19db, and also in covid19ks, which is my project that consumes all this stuff and generates the graphs of local interest to me at https://jgoerzen.github.io/covid19ks/

**Comment by cipriancraciun** [comment]
> I did adapt covid19db to download the various constituent datasets and combine them, and extract locations-diff.tsv from a previous commit, so I'm unblocked for now but it's a bit sub-optimal.

I've just made a commit and pushed the old files again (re-generated up-to-date).

However please note that in the following days I'll start removing some large non-archived files and replacing them with their `.gz` alternative.

Also I'll think about using an alternative distribution method.  I'll create a new issue for that and ping you there.

----

> As far as I can tell, GitHub actions have no particular limits for public repos on the free tier. But I may be missing something?

Apparently they did state that in some old documents, but removed them in the interim:
* https://web.archive.org/web/20200521202931/https://help.github.com/en/github/managing-large-files/what-is-my-disk-quota

----

> FWIW, the Rust code I'm using in covid19db runs in about 10 minutes in GitHub Actions. The majority of that time actually goes into compiling dependencies. The most time-consuming part of the actual data ingestion is in processing your datasets. Running the Rust code in debug mode (without optimizations) causes it to compile faster but run slower. It's probably not worth the effort, but I would surmise that redoing your processing pipeline in a compiled language like Rust or Go, targeting Sqlite directly with commits in long-running transactions, and then exporting the other formats from there would dramatically reduce the preparation time.

:)  I bet that using Rust, C/C++, or any more "performant" language, would make the processing blazing fast...  However at the moment when I wrote the scripts I used `jq` that allowed me to easily express various `map` / `reduce` pipelines, and some Python here and there.

At that time (1-2 months into the European situation) it ran on my laptop in a few minutes, which was acceptable...  However now, as months have started piling-up, it is quite a slug...  :)  (Time is not so much an issue, however the fan noise drives me mad...)  :)

Unfortunately rewriting all the scripts would take quite some time, and at the moment I don't think it is worth it.  (As you know, the "open-source" model doesn't pay the bills...)  :)

----

> Anyhow, I have made sure to credit your project in covid19db, and also in covid19ks, which is my project that consumes all this stuff and generates the graphs of local interest to me at https://jgoerzen.github.io/covid19ks/

Thanks!  I've also added your projects in my readme.

**Comment by cipriancraciun** [comment]
OK, bad news...  I had to remove the combined `values-sqlite.db.gz` variant, but the plain SQL inserts based on which it was created is still available as `values-sqlite.sql.gz`.

(The reason is that even compressed it surpassed the 100 MiB limit.)

**Comment by cipriancraciun** [comment]
Since I've hit the problem described in #46, I'm closing this issue (see that issue for details).
